spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Scheduler.scala

```diff
@@ -117,25 +152,39 @@ private[bigdl] class Scheduler[T] (
     node.element match {
       case s: SwitchOps[_] =>
         val switchNode = node.asInstanceOf[SwitchControlNode[Module[T]]]
-        selectNexts(switchNode.availableNodes(), node)
+        selectNexts(switchNode.availableNodes(), node, nextNodeFrame)
       case _ =>
-        selectNexts(node.nextNodes, node)
+        selectNexts(node.nextNodes, node, nextNodeFrame)
     }
   }
 
-  private def selectNexts(candidateNodes: Seq[ModuleNode[T]], curNode: ModuleNode[T]): Unit = {
+  private def startNextIteration(frame: Frame[T]): Unit = {
+    // Wake up the waiting nodes
+    frame.waitingNodes.foreach(readyQueue.enqueue(_))
+    frame.waitingNodes.clear()
+
+    // As frame is refreshed, mark all nodes in the frame are not ready and remove them
+    // from the frame
+    frame.nodes.filterNot(_.element.isInstanceOf[NextIteration[_, _]]).foreach(n => {
+      nodeStatus.unset(n)
+    })
```

Comment:
nodes in the should be removed according to the comment

Reply:
the comment is wrong. Fix the comment